src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2503,6 +2503,26 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return flags;
     }
 
+    til::point TermControl::_toControlOrigin(const til::point& terminalPos)
+    {
+        const til::size fontSize{ til::math::flooring, CharacterDimensions() };
+
+        // Convert text buffer cursor position to client coordinate position
+        // within the window. This point is in _pixels_
+        const til::point clientCursorPos{ terminalPos * fontSize };
+
+        // Get scale factor for view
+        const double scaleFactor = SwapChainPanel().CompositionScaleX();
+
+        const til::point clientCursorInDips{ til::math::flooring, clientCursorPos.x / scaleFactor, clientCursorPos.y / scaleFactor };
+
+        auto padding{ GetPadding() };
+        til::point relativeToOrigin{ til::math::flooring,
+                                     clientCursorInDips.x + padding.Left,
+                                     clientCursorInDips.y + padding.Top };

```

Comment:
I think padding is also in DIPs and would need to be added to the position before scaling it up.

Reply:
... I don't think so? I'm going from pixels to DIPs in `clientCursorPos / scale`, so that `relativeToOrigin` is just `dips + dips`

![image](https://user-images.githubusercontent.com/18356694/236005827-6a33079e-ad3c-4316-a437-2f50570ed476.png)